app/src/main/java/dgca/wallet/app/android/certificate/view/ViewCertificateFragment.kt

```diff
@@ -116,6 +127,19 @@ class ViewCertificateFragment : Fragment() {
         }
     }
 
+    private fun launchImageSharing() {
+        val fileForSharing = viewModel.certificate.value!!.qrCode.toFile(
```

Comment:
logic can be extracted to viewmodel. and in fragment we will observe event when file ready.
Also in this specific case add null check and Toast can be used to show error message.

val cert = viewModel.certificate.value
if (cert == null) {
    showError()
    return 
}

Reply:
Thank you! That definitely makes sense. Fixed that